build/ci/postInstall.js

```diff
@@ -125,7 +126,78 @@ function fixJupyterLabRenderers() {
     }
 }
 
+function fixJupyterLabFuture() {
+    const warnings = [];
+    [
+        'node_modules/@jupyterlab/services/lib/kernel/future.js',
+        'node_modules/@lumino/signaling/dist/index.es6.js'
+    ].forEach((file) => {
+        const filePath = path.join(__dirname, '..', '..', file);
+        if (!fs.existsSync(filePath)) {
+            return;
+        }
+        const textToReplace = `return ok ? requestAnimationFrame : setImmediate;`;
+        const textToReplaceWith = `return ok ? requestAnimationFrame : (typeof setImmediate === 'function' ? setImmediate : setTimeout);`;
+        const fileContents = fs.readFileSync(filePath, 'utf8').toString();
+        if (fileContents.indexOf(textToReplace) === -1 && fileContents.indexOf(textToReplaceWith) === -1) {
+            warnings.push(`Unable to find Jupyter kernel/future/setImmediate usage to replace! in ${file}`);
+        }
+        fs.writeFileSync(filePath, fileContents.replace(textToReplace, textToReplaceWith));
+    });
+    if (warnings.length === 2) {
+        throw new Error(warnings[0] + '\n' + warnings[1]);
+    }
+}
+function fixLuminoPolling() {
+    const warnings = [];
+    ['node_modules/@lumino/polling/dist/index.es6.js'].forEach((file) => {
+        const filePath = path.join(__dirname, '..', '..', file);
+        if (!fs.existsSync(filePath)) {
+            return;
+        }
+        const textToReplace = `: setImmediate;`;
+        const textToReplaceWith = `: typeof setImmediate === 'function' ? setImmediate : setTimeout;`;
```

Comment:
Will submit PR upstream, `setImmediate` only runs in node. however we run our extensions in webworker, and jupyter code is written to run either in browser or node.

Reply:
will look into (tomorrow) using the setimmediate pollyfill npm package as an alternative to modifying such scripts.